guide/03-the-gis/accessing-and-managing-groups.ipynb

```diff
@@ -1 +1,896 @@
adata": {}, "outputs": [{"data": {"text/plain": ["{'owner': 'arcgis_python',\n", " 'admins': ['arcgis_python'],\n", " 'users': ['api_data_owner']}"]}, "execution_count": 27, "metadata": {}, "output_type": "execute_result"}], "source": ["geocaching_group.get_members()"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Updating a group\n", "You can update any or all the fields of a group that you specified when creating it. Thus, the `update()` accepts the same parameters as `create()`. Let us close the group to members who can be added through invitation only. This prevents users from sending a joining request."]}, {"cell_type": "code", "execution_count": 28, "metadata": {}, "outputs": [{"data": {"text/plain": ["True"]}, "execution_count": 28, "metadata": {}, "output_type": "execute_result"}], "source": ["geocaching_group.update(is_invitation_only=True)"]}, {"cell_type": "code", "execution_count": 29, "metadata": {}, "outputs": [{"data": {"text/plain": ["True"]}, "execution_count": 29, "metadata": {}, "output_type": "execute_result"}], "source": ["geocaching_group.isInvitationOnly"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Terminating a group\n", "If a group no longer serves the purpose or if its requirements have changed, the owner and members have a few options. They can leave the group by calling the `leave()` method. When users leave a group, the content they shared with the group will automatically be unshared. The owner or org admin can delete the group by calling the `delete()` method. \n", "\n", "However if the contents and the group have to be preserved, the group's ownership can be transferred by calling the `reassign_to()` and specifying the new owner's username. \n", "\n", "<blockquote><b>Note:</b> Only the administrator of the org can reassign a group to another user.</blockquote>"]}, {"cell_type": "markdown", "metadata": {}, "source": ["let us reassign ownership to another user\n", "```python\n", "geocaching_group.reassign_to(target_owner = 'api_data_owner')\n", "```"]}, {"cell_type": "markdown", "metadata": {}, "source": ["Finally, let us delete this group"]}, {"cell_type": "code", "execution_count": 35, "metadata": {}, "outputs": [{"data": {"text/plain": ["True"]}, "execution_count": 35, "metadata": {}, "output_type": "execute_result"}], "source": ["geocaching_group.delete()"]}], "metadata": {"kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.8.2"}, "toc": {"base_numbering": 1, "nav_menu": {}, "number_sections": false, "sideBar": false, "skip_h1_title": true, "title_cell": "Table of Contents", "title_sidebar": "Contents", "toc_cell": true, "toc_position": {}, "toc_section_display": true, "toc_window_display": false}}, "nbformat": 4, "nbformat_minor": 1}
```

Comment:
<p>We'd like to shy away from explaining too much information in code cell comments, but rather use code or a markdown cell. In this case, I think it'd be better to delete the csv item first. Add a sentence explaining and then show how to delete it:</p><p> </p><p><em>We're going to be create a CSV item called Hidden treasures, but before we do, let's ensure we don't already own on is this portal. If we do, we'll delete it so the subsequent code will work.</em></p><p> </p><pre class="ql-syntax" spellcheck="false">for csv_item in gis.content.search(f"title: Hidden treasures and username:{username)"):
   try:
       csv_item.delete()
   except Exception as e:
       pass  
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1633/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>